### PR TITLE
Trenches - Terrain Trenching

### PR DESCRIPTION
**Pull request by PabstMirror**
very rough draft

https://www.youtube.com/watch?v=Qfz-AD6txNc

dig trenches in to the ground
- ONLY straight east/north

yuck
we drop the terrain on the red dots
then place these terrain blocks in the green to create the edge we want

### CODE REVIEW

**Comment by Drofseh** [comment]
Do you think it'd be worth adding grass cutters fish the centre of the trench?
There aren't normally plants in the bottom of recently dug trenches lol

**Comment by 654wak654** [comment]
> Do you think it'd be worth adding grass cutters to the centre line of the trench? There aren't normally plants in the bottom of recently dug trenches lol

image

**Comment by 654wak654** [comment]
> Do you think it'd be worth adding grass cutters to the centre line of the trench? There aren't normally plants in the bottom of recently dug trenches lol

image

**Comment by veteran29** [comment]
Grass cutters mess with AI pathing IIRC, but trench objects most likely do so too.

**Comment by veteran29** [comment]
Grass cutters mess with AI pathing IIRC, but trench objects most likely do so too.

**Comment by PabstMirror** [comment]
added a third block for the floor, so it's a nice dirt texture and flat bottom
added option to run grass cutters (but I think a little clutter helps theses look less blocky)

test

File: addons/zeus/functions/fnc_moduleLayTrench.sqf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    [LSTRING(RequiresAddon)] call FUNC(showMessage);
+};
+
+private _startPos = getPosASL _logic; // change position of logic to be aligned with grid
+getTerrainInfo params ["", "", "_cellsize"];
+_startPos = (_startPos select [0,2]) apply {_cellsize * round (_x / _cellsize)};
+_logic setPos _startPos;
+
+
+// todo: it would be nice to show visually that trenches can only be dug north/south or east/west
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleLayTrench.sqf:37]: Put it in the tooltip of the module?

**Review by jonpas — APPROVED** [review]

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**